app/controllers/v1/submissions_controller.rb

```diff
@@ -13,7 +13,7 @@ def create
     task = Task.find(params.dig(:submission, :task_id))
 
     if !correcting_submission? && task.active_submission&.completed?
-      render jsonapi: task.active_submission, status: :ok
+      render jsonapi_errors: [{ title: 'Task already has completed submission.' }], status: :bad_request
```

Comment:
Could we use something other than Bad Request? This [400](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/400) status (4xx – user error) is meant to mean "your request was malformed, please correct your request format and try again". Their request format was probably fine and we don't want them to try again, we want to tell them "hey, the state we have on the thing you're trying to change doesn't match what you're trying to do". [409 Conflict](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/409) might be a better bet?

Reply:
Fair. Done.